src/core/parseHTML.js

```diff
@@ -15,7 +15,7 @@ define([
 // defaults to document
 // keepScripts (optional): If true, will include scripts passed in the html string
 jQuery.parseHTML = function( data, context, keepScripts ) {
-	if ( !data || typeof data !== "string" ) {
+	if ( typeof data !== "string" ) {
 		return null;
 	}
```

Comment:
I'd even support a followup to drop this `if` entirely. If non-string input really causes a problem, it should be visible as an exception.